Scripts/linux/dtllabservice.py

```diff
@@ -332,55 +332,128 @@ def getVirtualMachine(self, subscriptionId, vmId=None, name=None):
+                "size": "Standard_DS2"
             }
         ]
         """
 
         if vmId is not None:
-            fieldName = 'Id'
+            fieldName = 'computeId'
             fieldValue = vmId
         else:
-            fieldName = 'Name'
+            fieldName = 'name'
             fieldValue = name
 
-        url = '/subscriptions/{0}/providers/microsoft.devtestlab/environments/?$filter=tolower({1})%20eq%20tolower(%27{2}%27)&api-version={3}'.format(
+        url = '/subscriptions/{0}/providers/microsoft.devtestlab/environments/?api-version={1}'.format(
             subscriptionId,
-            fieldName,
-            fieldValue,
             self._apiVersion
         )
 
         api = azurerest.AzureRestHelper(self._settings, self._settings.accessToken, self._host)
-        return api.get(url, self._apiVersion)
+        environments = api.get(url, self._apiVersion)
+
+        if environments is None:
+            return None
+
+        environments = environments['value']
+        vms = []
+
+        for environment in environments:
+            for vm in environment['properties']['vms']:
+                if vm[fieldName].lower() == fieldValue.lower():
+                    vm['environmentId'] = environment['id']
+                    vms.append(vm)
+                    continue
+
+        return vms
+
+    def deleteVirtualMachine(self, subscriptionId, labName, name=None, vmId=None):
+        """Deletes the environment including virtual machine of the environment with the specified name in the specified
+           lab.  If there is more than one environment with name in labName, an error will be thrown.
+
+        Args:
+            subscriptionId (string) - The subscription in which to filter.
+            labName (string)        - The ID of the lab in which to filter.
+            name (string)           - The name of tha virtual machine.  If there are more than one virtual machine with
+                                      the same name in the specified lab, you must use the vmId parameter instead.
+            vmId (string)           - The resource ID of the virtual machine.
+        Returns:
+            0 if successful, 1 otherwise.
+        """
+
+        vms = self.getVirtualMachine(subscriptionId, name, vmId)
+
+        if vms is None or len(vms) == 0:
+            self._printService.error('No virtual machines found.')
+            return 1
+        elif vms is not None and len(vms) > 1:
+            self._printService.dumps(vms)
+            self._printService.error(
+                'More than one virtual machine named {0} found in the lab.  Use virtual machine ID instead'.format(
+                    name))
+            return 1
+
+        vm = vms[0]
+
+        self._printService.info('Deleting virtual machine: ' + vm['name'])
+
+        environmentId = vm['environmentId']
+
+        url = environmentId + '?api-version={0}'.format(self._apiVersion)
```

Comment:
This might be a tad more readable:
`url = '{0}?api-version{1}'.format(environmentId, self._apiVersion)`